packages/extension/src/serviceWorker/createCookieObject.ts

```diff
@@ -152,9 +186,21 @@ function parseAttributeValues(
     }
     value = ((value || '') as string).toLowerCase();
   }
+  if (type === 'size' && !value) {
+    const encoder = new TextEncoder();
+    value = encoder.encode(cookieValue).length;
+  }
 
   if (type === 'expires' && value !== 0) {
-    value = new Date(value as string).toJSON() || 'Session';
+    value = calculateEffectiveExpiryDate(value as string) || 0;
+  }
+
+  if (type === 'priority' && !value) {
+    value = 'Medium';
+  }
+
+  if (type === 'partitionKey' && !value) {
```

Comment:
Too many if conditions, I think it needs some refactoring. Probably something like this,


```js
let value = parsedCookieValue || chromeStoreCookieValue || prevParsedCookieValue;

  switch (type) {
    case 'domain':
      if (url) {
        value = value || new URL(url).hostname;
      } else {
        value = value || '';
      }
      break;

    case 'path':
      value = value || '/';
      break;

    case 'secure':
    case 'httponly':
      value = value || false;
      break;

    case 'samesite':
      if (value === 'no_restriction') {
        value = 'none';
      } else if (value === 'unspecified') {
        value = '';
      }
      value = (value || '').toLowerCase();
      break;

    case 'size':
      if (!value) {
        const encoder = new TextEncoder();
        value = encoder.encode(cookieValue || '').length;
      }
      break;

    case 'expires':
      if (value !== 0) {
        value = calculateEffectiveExpiryDate(value as string) || 0;
      }
      break;

    case 'priority':
      value = value || 'Medium';
      break;

    case 'partitionKey':
      value = value || '';
      break;
  }

  return value;
```